### PR TITLE
Bump package versions

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
None

## Summary
Bumping versions

### CODE REVIEW

**Comment by brokensound77** [comment]
Is a bump to [this](https://github.com/elastic/detection-rules/blob/b42725aba816694fe600e2e761be96d45d450d01/etc/packages.yml#L27-L29) or any other change needed in the registry settings @rw-access?

**Comment by rw-access** [comment]
Yeah, @brokensound77 that too.
Also, make sure you double check the "backport" label on this PR. I manually set it to `backport: skip`, since we don't want this change going to 7.13 and 7.14 branches.

File: detection_rules/schemas/__init__.py [hunk 1]
```diff
@@ -199,7 +205,7 @@ def get_stack_schemas(stack_version: str) -> Dict[str, dict]:
     versions = {k: v for k, v in stack_map.items()
                 if (mapped_version := Version(k)) >= stack_version and mapped_version <= current_package and v}
 
-    if stack_version > current_package:
+    if stack_version > current_package or not versions:
```
Comment (rw-access) [inline detection_rules/schemas/__init__.py:208]: this seems like it would mask important errors
```suggestion
    if stack_version > current_package:
```

Comment (brokensound77) [reply detection_rules/schemas/__init__.py]: Good point. I restored it and replaced it with a test b53a9bf and a 7.15 entry with a `TODO`

File: tests/test_schemas.py [hunk 2]
```diff
@@ -194,3 +195,14 @@ def build_rule(query):
...
+
+class TestVersions(unittest.TestCase):
+    """Test that schema versioning aligns."""
+
+    def test_stack_schema_map(self):
+        """Test to ensure that an entry exists in the stack-schema-map for the current package version."""
+        package_version = load_current_package_version()
+        stack_map = utils.load_etc_dump('stack-schema-map.yaml')
+        err_msg = f'There is no entry defined for the current package ({package_version}) in the stack-schema-map'
+        self.assertIn(package_version, ['.'.join(v.split('.')[:2]) for v in stack_map], err_msg)
```
Comment (rw-access) [inline tests/test_schemas.py:208]: this is a little cryptic. maybe... this?
```suggestion
        self.assertIn(package_version, [str(Version(v)[:2]) for v in stack_map], err_msg)
```

File: etc/packages.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 package:
-  name: "7.14"
+  name: "7.15"
```
Comment (rw-access) [inline etc/packages.yml:3]: shoot, i think we have to split this in two PRs now.
one without this file that does a backport for the code changes.

and one that does backport the code changes.

another option: we mark this file exempt from backports:
https://github.com/elastic/detection-rules/blob/main/detection_rules/devtools.py#L142-L149

Comment (brokensound77) [reply etc/packages.yml:3]: I do like the idea of exempting it to keep us safe from any accidental backports of this file in the future. Additional idea: add a job to post a comment when the backport label added and an exempted change is included

Comment (rw-access) [reply etc/packages.yml:3]: > Additional idea: add a job to post a comment when the backport label added and an exempted change is included

Yeah, I think this is a good long term improvement. While the PR is still open, comment with the files that are exempted from the backport. Then it's less hidden and magical. So basically run `--dry-run` and paste that in.

Until then, what would you prefer to do within this PR? Split in two or update the backport logic to exempt the file?

Or should I make a quick PR that we merge first that will exempt etc/package.yml?

Comment (brokensound77) [reply etc/packages.yml:3]: It would be a good real test, so if you have the cycles, making the PR and merging would be my preference

**Comment by brokensound77** [comment]
Do we need to save api-schema for 7.14 as part of this too @rw-access

File: etc/packages.yml (hunk 3 above)
Comment (rw-access) [reply etc/packages.yml:3]: PR to add the exemption
https://github.com/elastic/detection-rules/pull/1437

**Review by rw-access — APPROVED** [review]

**Pull request merged**